st2api/st2api/controllers/v1/actionexecutions.py

```diff
@@ -515,22 +515,48 @@ def put(self, id, liveaction_api, requester_user, show_secrets=False):
             abort(http_client.INTERNAL_SERVER_ERROR,
                   'Execution object missing link to liveaction %s.' % liveaction_id)
 
-        if liveaction_db.status in LIVEACTION_COMPLETED_STATES:
+        if liveaction_db.status in action_constants.LIVEACTION_COMPLETED_STATES:
             abort(http_client.BAD_REQUEST, 'Execution is already in completed state.')
 
+        if (getattr(liveaction_api, 'result', None) is not None and
+                liveaction_api.status in [
+                    action_constants.LIVEACTION_STATUS_PAUSING,
+                    action_constants.LIVEACTION_STATUS_PAUSED,
+                    action_constants.LIVEACTION_STATUS_RESUMING]):
+            abort(http_client.BAD_REQUEST,
+                  'The result is not applicable for pausing and resuming execution.')
+
         try:
-            liveaction_db = action_service.update_status(
-                liveaction_db,
-                liveaction_api.status,
-                result=getattr(liveaction_api, 'result', None)
-            )
+            if (liveaction_api.status == action_constants.LIVEACTION_STATUS_PAUSING or
+                    liveaction_api.status == action_constants.LIVEACTION_STATUS_PAUSED):
+                liveaction_db, actionexecution_db = action_service.request_pause(
```

Comment:
So I assume we intentionally allow user to request pause multiple times, even if the execution is in one of those states?

Aka because operation is idempotent or to allow user to "rectify" the situation in case something gets stuck / fails half way through?